crux-mir-comp/DESIGN.md

```diff
@@ -147,6 +147,55 @@ implementations of some cryptographic algorithms) are usually declared as
 immutable and thus always retain their initial values.
 
 
+## Cryptol
+
+It is possible to import Cryptol expressions into symbolic Rust test cases to
+help write specifications (e.g., if you want to check that a Rust implementation
+matches a Cryptol specifications).  This is done by using the `cryptol!` macro
+defined in the `crucible` package.  Here's an example illustrating how to
+import a Cryptol function `myCryFun` defined in Cryptol module `SomeCryMod`,
+and bind it as the Rust function `myRustFun`:
+
+```Rust
+use crucible::*;
+cryptol! {
+    path "SomeCryMod";
+    pub fn myRustFun(x: u8) -> u32 = r"myCryFun";
+}
+```
+
+Тhe `path` component of the macro specifies a Cryptol module,
+the string after the `=` is a Cryptol expressions, and the rest of the
+declaration specifies how to invoke evaluating the Cryptol expression from
+Rust.  If the declaration contains `const` generics, than the Cryptol expression
+is specified as a format string which may refer to the values of the const
+generic parameters in curly braces (literal curly braces need to be escaped
+as a double curly brace).
```

Comment:
It would be worth showing a concrete example of what an imported const generic function looks like, as the syntax is somewhat non-obvious.